### PR TITLE
SK Fix Resave and Fixes Standardisation Pipeline

### PR DESCRIPTION
**Pull request by scarlett-k-nhs**
# Summary

**What/Why**

Add your summary here - keep it brief, to the point, and in plain English. [For further
information about pull requests, check out the GDS
Way](https://gds-way.cloudapps.digital/standards/pull-requests.html).

**Your approach**

Why did you choose this approach?

**What to review**

Include any helpful pointers for the reviewer.

# Checklists

This pull/merge request meets the following requirements:

- [ ] code runs
- [ ] [developments are ethical][data-ethics-framework]
- [ ] developments are secure (no data or notebook output)
- [ ] you have made proportionate checks that the code works correctly
- [ ] [minimum usable documentation][agilemodeling] written in the `docs` folder

Comments have been added below around the incomplete checks.

[agilemodeling]: http://agilemodeling.com/essays/documentLate.htm
[data-ethics-framework]: https://www.gov.uk/government/publications/data-ethics-framework

### CODE REVIEW

File: src/extraction/extraction.py [hunk 1]
```diff
@@ -81,13 +81,15 @@ def run_or_load(
         self,
         verbose: bool = False,
         resave: bool = False,
```
Comment (danjscho) [inline src/extraction/extraction.py:83]: Would this be clearer as `overwrite`?

File: src/extraction/extraction.py [hunk 2]
```diff
@@ -113,8 +115,9 @@ def run_or_load(
                 verbose=verbose,
             )
 
-            if resave or file_exists(self.path_output) is False:
-                save_json(data=patients_entities, path=self.path_output)
+            if save:
+                if resave or file_exists(self.path_output) is False:
```
Comment (danjscho) [inline src/extraction/extraction.py:119]: Possible stylistic change, instead of `if a or b is False` use `if a or not b`?

**Review by danjscho — APPROVED** [review]
Couple of thoughts added, but looks good either way.

File: src/generate/llm.py [hunk 3]
```diff
@@ -83,8 +87,9 @@ def run_or_load(
 
             output = chain.batch(batch)
 
-            if resave or file_exists(self.path_output) is False:
-                save_json(output, self.path_output)
+            if save:
+                if overwrite or file_exists(self.path_output) is False:
```
Comment (danjscho) [inline src/generate/llm.py:91]: Just noted another bool to change

**Review by danjscho — APPROVED** [review]

**Review by willpoulett — APPROVED** [review]

**Pull request merged**